libyul/AST.h

```diff
@@ -35,49 +35,67 @@ namespace solidity::yul
 
 using Type = YulString;
 
-struct TypedName { langutil::SourceLocation location; YulString name; Type type; };
+struct DebugData
+{
+	explicit DebugData(langutil::SourceLocation _location) : location(std::move(_location)) {}
+	langutil::SourceLocation location;
+};
+
+inline std::shared_ptr<DebugData const> createDebugData(langutil::SourceLocation _location = {})
+{
+	return std::make_shared<DebugData const>(_location);
+}
+
+inline void changeDebugDataLocationEnd(std::shared_ptr<DebugData const>& io_debugData, int _end)
```

Comment:
Isn't this really only used in `AsmParser`? Wouldn't we better off moving this helper in `AsmParser` and potentially calling up `updateLocationEnd()`?

Reply:
If it is used only in AsmParser it makes sense to move it over there and rename. Will check that

Reply:
Moved to `AsmParser` and renamed to `updateLocationEnd `

Reply:
Well, it should not be exposed as a standalone helper but how about putting it inside `DebugData`? I think it would make sense there as something like this:
```c++
shared_ptr<DebugData> DebugData::withEndAt(SourceLocation const& _location) const;
```